<launch>

	<arg name="rviz" default="true" />
	<arg name="lidar" default="ouster" /> <!-- ouster, velodyne, livox, kitti, ... -->
	<arg name="odom_topic" default="/Odometry" />
	<arg name="lidar_topic" default="/cloud_registered" />

	<!-- visualize -->
	<!-- <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rviz_lio" args="-d $(find fast_lio_sam_qn)/config/lio_rviz.rviz" launch-prefix="nice"/> -->
	<node if="$(arg rviz)" pkg="rviz" type="rviz" name="rviz_sam" args="-d $(find fast_lio_sam_qn)/config/sam_rviz.rviz" launch-prefix="nice"/>

	<!-- run fast_lio_sam_qn -->
	<rosparam command="load" file="$(find fast_lio_sam_qn)/config/config.yaml" />
	<node pkg="fast_lio_sam_qn" type="fast_lio_sam_qn_node" name="fast_lio_sam_qn_node" output="screen">
		<remap from="/Odometry" to="$(arg odom_topic)"/>
		<remap from="/cloud_registered" to="$(arg lidar_topic)"/>
	</node>

	<!-- run fast_lio -->
	<include if="$(eval arg ('lidar') == 'ouster')" file="$(find fast_lio)/launch/mapping_ouster64.launch">
		<arg name="rviz" value="false"/>
	</include>
	<include if="$(eval arg ('lidar') == 'velodyne')" file="$(find fast_lio)/launch/mapping_velodyne.launch">
		<arg name="rviz" value="false"/>
	</include>
	<include if="$(eval arg ('lidar') == 'livox')" file="$(find fast_lio)/launch/mapping_avia.launch">
		<arg name="rviz" value="false"/>
	</include>
	<include if="$(eval arg ('lidar') == 'kitti')" file="$(find fast_lio)/../fastlio_config_launch/kitti.launch">
		<arg name="rviz" value="false"/>
	</include>
	<include if="$(eval arg ('lidar') == 'mulran')" file="$(find fast_lio)/../fastlio_config_launch/mulran.launch">
		<arg name="rviz" value="false"/>
	</include>
	<include if="$(eval arg ('lidar') == 'newer-college2020')" file="$(find fast_lio)/../fastlio_config_launch/newer-college2020.launch">
		<arg name="rviz" value="false"/>
	</include>
	<include if="$(eval arg ('lidar') == 'kimera-multi-acl_jackal2')" file="$(find fast_lio)/../fastlio_config_launch/kimera-multi-acl_jackal2.launch">
		<arg name="rviz" value="false"/>
	</include>
	<include if="$(eval arg ('lidar') == 'kimera-multi-apis')" file="$(find fast_lio)/../fastlio_config_launch/kimera-multi-apis.launch">
		<arg name="rviz" value="false"/>
	</include>
	<include if="$(eval arg ('lidar') == 'vbr-colosseo')" file="$(find fast_lio)/../fastlio_config_launch/vbr-colosseo.launch">
		<arg name="rviz" value="false"/>
	</include>

</launch>
